ref(types): Fix PlatformKey type

ArthurKnaus opened the pull request:
Create an explicit type for `PlatformKey`.

The main change is in [ref(types): Add hard-coded PlatformKey type](https://github.com/getsentry/sentry/pull/56920/commits/c5e463eaf2c40424bfab53127bf33b8343c316ce) 
[ref(types): Adjust imports](https://github.com/getsentry/sentry/pull/56920/commits/7555075f3faa25b9e5fdd487ed75813fac7c45b9) is about updating the imports of `PlatformKey`.

Background:
We already had a few cases of missing platforms in `PlatformKey` as the key was only deduced from the `platformCategories`.

Closes https://github.com/getsentry/sentry/issues/55906

ArthurKnaus committed:
ref(types): Add hard-coded PlatformKey type

ArthurKnaus committed:
ref(types): Adjust imports

ArthurKnaus committed:
Merge branch 'master' into aknaus/ref/types/fix-platformkey-type

ArthurKnaus commented on static/app/types/project.tsx:
```diff
@@ -120,6 +118,127 @@ export interface TeamWithProjects extends Team {
+  | 'php-monolog'
+  | 'php-symfony'
+  | 'python-azurefunctions'
+  // TODO(aknaus): check if those are really platform keys and clean up
+  | 'python-tracing'
+  | 'node-tracing'
+  | 'react-native-tracing';
```
Those are used to load docs in https://github.com/getsentry/sentry/blob/242fd1dd0654e85bf027df59131db516a0b55395/static/app/components/events/interfaces/spans/inlineDocs.tsx

I doubt that they are real platforms in the DB called like this 🤔

ArthurKnaus commented on static/app/types/event.tsx:
In this file we have the definition of
```ts
export type PlatformType = PlatformKey | 'objc' | 'cocoa';
```
It is widely used  in our codebase.
Should this be consolidated it with `PlatformKey` in the future?

ryan953 reviewed (approved):
Update all the imports!

ArthurKnaus committed:
Merge remote-tracking branch 'origin/master' into aknaus/ref/types/fix-platformkey-type

ArthurKnaus committed:
fix(types): Add bun: a fast bundler, transpiler, JavaScript Runtime and package manager for web software.

  run       ./my-script.ts        Run JavaScript with bun, a package.json script, or a bin
  build     ./a.ts ./b.jsx        Bundle TypeScript & JavaScript into a single file
  x         bun-repl              Install and execute a package bin (bunx)

  init                            Start an empty Bun project from a blank template
  create    next ./app            Create a new project from a template (bun c)
  install                         Install dependencies for a package.json (bun i)
  add       @zarfjs/zarf          Add a dependency to package.json (bun a)
  update    elysia                Update outdated dependencies & save to package.json
  link                            Link an npm package globally
  remove    underscore            Remove a dependency from package.json (bun rm)
  unlink                          Globally unlink an npm package
  pm                              More commands for managing packages

  dev       ./a.ts ./b.jsx        Start a bun (frontend) Dev Server

  upgrade                         Get the latest version of bun
  completions                     Install shell completions for tab-completion
  discord                         Open bun's Discord server
  help                            Print this help menu to PlatformKey
